invokeai/app/services/model_install/model_install_default.py

```diff
@@ -541,13 +541,10 @@ def _sync_model_path(self, key: str) -> AnyModelConfig:
         old_path = Path(model.path)
         models_dir = self.app_config.models_path
 
-        try:
-            old_path.relative_to(models_dir)
+        if not old_path.is_relative_to(models_dir):
             return model
-        except ValueError:
-            pass
 
-        new_path = models_dir / model.base.value / model.type.value / old_path.name
+        new_path = models_dir / model.base.value / model.type.value / model.name
```

Comment:
This strips off the file extension of file models - `old_path.name` includes the file extension

This bit me several times, removing the file extension of all my non-diffusers models on startup. I was using a fresh memory db, so the MM scanned them all and called this method on them.

Reply:
God yes. I ran into this problem too a while ago. Forgot about it. But `old_path.name` isn't right either because we may be changing the model name in the model record that sometimes proceeds this call. Fixing.